utils/src/main/scala/it/pagopa/interop/commons/utils/errors/AkkaResponses.scala

```diff
@@ -6,21 +6,29 @@ import akka.http.scaladsl.server.StandardRoute
 import cats.implicits.toFunctorFilterOps
 import com.typesafe.scalalogging.LoggerTakingImplicit
 import it.pagopa.interop.commons.logging.ContextFieldsToLog
+import it.pagopa.interop.commons.utils.CORRELATION_ID_HEADER
 import it.pagopa.interop.commons.utils.errors.GenericComponentErrors.GenericError
 
 trait AkkaResponses {
 
   private def completeWithError(statusCode: StatusCode, error: ComponentError)(implicit
+    contexts: Seq[(String, String)],
     serviceCode: ServiceCode
-  ): StandardRoute = complete(statusCode.intValue, Problem(statusCode, error, serviceCode))
+  ): StandardRoute = complete(statusCode.intValue, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
 
   private def completeWithErrors(statusCode: StatusCode, errors: List[ComponentError])(implicit
+    contexts: Seq[(String, String)],
     serviceCode: ServiceCode
-  ): StandardRoute = complete(statusCode.intValue, Problem(statusCode, errors, serviceCode))
+  ): StandardRoute = complete(statusCode.intValue, Problem(statusCode, errors, serviceCode, getCorrelationId(contexts)))
 
   private def completeWithError(statusCode: StatusCode, headers: List[HttpHeader], error: ComponentError)(implicit
+    contexts: Seq[(String, String)],
     serviceCode: ServiceCode
-  ): StandardRoute = complete(statusCode.intValue, headers, Problem(statusCode, error, serviceCode))
+  ): StandardRoute =
+    complete(statusCode.intValue, headers, Problem(statusCode, error, serviceCode, getCorrelationId(contexts)))
+
+  private def getCorrelationId(contexts: Seq[(String, String)]): Option[String] =
```

Comment:
I'll maybe add @inline to speed it up a little

Reply:
done